arangod/Pregel/PregelMetrics.cpp

```diff
@@ -0,0 +1,83 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Markus Pfeiffer
+////////////////////////////////////////////////////////////////////////////////
+
+#include "PregelMetrics.h"
+
+#include "Pregel/PregelMetricsDeclarations.h"
+#include "Metrics/MetricsFeature.h"
+
+using namespace arangodb::pregel;
+
+PregelMetrics::PregelMetrics(metrics::MetricsFeature& metricsFeature)
+    : ReplicatedLogMetrics(&metricsFeature) {}
+
+template<typename Builder, bool mock>
+auto PregelMetrics::createMetric(metrics::MetricsFeature* metricsFeature)
+    -> std::shared_ptr<typename Builder::MetricT> {
+  TRI_ASSERT((metricsFeature == nullptr) == mock);
+  if constexpr (!mock) {
```

Comment:
This mock looks odd to me. Shouldn't it be possible to create a Mock metrics feature, which would have the function addShared do the same thing as in line 41? But I see that this is how it was done already in ReplicationLogMetrics, so perhaps it is more difficult.